src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -120,10 +146,60 @@ protected String name() {
     }
 
     public static class LayoutUpdateVisitor extends ViewVisitor {
-        public LayoutUpdateVisitor(List<Pathfinder.PathElement> path, LayoutRule args) {
+        private class CycleDetector {
```

Comment:
This could be static (and we usually lay out our files)
```
public static members

public members (rare!)

public inner classes

public constructors

public methods

private methods

private classes

private members

private static members
```